taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2147,8 +2146,34 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
             alloca_type->getArrayNumElements() == 1 &&
             alloca_type->getArrayElementType() == char_type) {
           auto new_type = llvm::ArrayType::get(char_type, stmt->tls_size);
-          alloca->setAllocatedType(new_type);
-          replaced_alloca_types += 1;
+          {
+            llvm::IRBuilderBase::InsertPointGuard guard(*builder);
+            builder->SetInsertPoint(alloca);
+            auto *new_alloca = builder->CreateAlloca(new_type);
+            new_alloca->setAlignment(Align(8));
+            replaced_alloca_types += 1;
+            for (llvm::User *user : alloca->users()) {
+              if (llvm::GetElementPtrInst *gep =
+                      llvm::dyn_cast<llvm::GetElementPtrInst>(user)) {
+                if (gep->getPointerOperand() == alloca) {
+                  std::vector<Value *> indices(gep->idx_begin(),
+                                               gep->idx_end());
+                  {
+                    builder->SetInsertPoint(gep);
+                    auto *new_gep =
+                        builder->CreateGEP(new_type, new_alloca, indices);
+                    llvm::cast<llvm::GetElementPtrInst>(new_gep)->setIsInBounds(
+                        true);
+                    gep->replaceAllUsesWith(new_gep);
+                    gep->eraseFromParent();
+                    break;
+                  }
+                }
+              }
+            }
+            alloca->eraseFromParent();
```

Comment:
Are there users other than GEP?

Reply:
Oh maybe I can simply loop over the users of the AllocaInst instead of looping over all the instructions in the function. Thanks.